File: rotatestring.js
//Objective is to see whether two strings are rotations of each other.

let str1 = 'ABCDE'
let str2 = 'CDEAB'


//O(n * m) solution that checks if the two strings are of same length and
//the substring of one exists in the other.

return str1.length == str2.length && (str1 + str1).indexOf(str2) !== -1

File: rotatestring.java
//Java Solution

class Solution {
    public boolean rotateString(String A, String B) {
        if (A.length() != B.length()) {
            return false;
        }
        
        if ((A + A).indexOf(B) != -1) {
            return true;
        }
        
        return false;
    }
}